### PR TITLE
Support editing task priorities

### LINKED ISSUE #449: Support Editing Priority
### A detailed description of the feature you would like to see added.

TL;DR: I would like to be able to edit the priority of a task within the queue.

The recent release of pueue added support for a `--priority` flag to the `pueue add` command. It would be great to be able to also edit the priority.

### Explain your usecase of the requested feature

For example, if I have recently added an item to the queue but forgot to set it to a high priority, I would be able to execute `pueue edit --priority 9 126`.

### Alternatives

At the moment, the only alternative I can think of is to remove and re-add the task with a higher priority.

### Additional context

_No response_

### PR DESCRIPTION
**Pull request by jch-13**
Adds support for the modification of task priorities in a queue via `pueue edit --priority/-o` and `pueue restart --edit-priority/-o` (fixes #449). Despite the interesting discussion in #449 about how to improve task editing in general, this takes the easy way out and tries to integrate with the existing editing system by using (multiple) editor instances for now. 

## Checklist

- [x] I picked the correct source and target branch.
- [x] I included a new entry to the `CHANGELOG.md`.
- [x] I checked `cargo clippy` and `cargo fmt`. The CI will fail otherwise anyway.
- [x] (If applicable) I added tests for this feature or adjusted existing tests.
- [x] (If applicable) I checked if anything in the wiki needs to be changed.

### CODE REVIEW

File: CHANGELOG.md [hunk 1]
```diff
@@ -8,6 +8,8 @@ The format is based on [Keep a Changelog](https://keepachangelog.com/en/1.0.0/),
 
 ### Added
 
+- Support modification of task priorities via `pueue edit --priority/-o` and `pueue restart --edit-priority/-o` [#449](https://github.com/Nukesor/pueue/issues/449)
```
Comment (Nukesor) [inline CHANGELOG.md:11]: Minor merge conflict.

I forgot to update the development branch after the latest patch releases on master again. It's an easy fix though :)

File: pueue/tests/client/integration/edit.rs [hunk 2]
```diff
@@ -94,6 +95,33 @@ async fn edit_delete_label() -> Result<()> {
...
+    send_message(shared, message)
+        .await
+        .context("Failed to to add stashed task.")?;
+
+    // Echo a new priority into the file.
+    let mut envs = HashMap::new();
+    envs.insert("EDITOR", "echo '99' > ");
+    run_client_command_with_env(shared, &["edit", "--priority", "0"], envs)?;
+
+    // Make sure that the label has indeed be deleted
```
Comment (Nukesor) [inline pueue/tests/client/integration/edit.rs:117]:
```suggestion
    // Make sure that the priority has indeed been updated.
```

**Review by Nukesor — CHANGES_REQUESTED** [review]
Thanks for the PR :) 

It looks pretty clean, two minor nitpicks. I kind of let the feature stall since I wanted to fix this in a proper way, but I guess you're right. This is an good intermediate solution until there's a conclusion on how to move forward.

**Comment by jch-13** [comment]
Thank you! :)

**Comment by Nukesor** [comment]
Thanks :)

**Pull request merged**